Lib/Schema/Field.php

```diff
@@ -0,0 +1,662 @@
+<?php
+/**
+ * Field class file.
+ *
+ * @package Freyja\Database\Schema
+ * @copyright 2016 SqueezyWeb
+ * @since 0.1.0
+ */
+
+namespace Freyja\Database\Schema;
+
+use Freyja\Exceptions\InvalidArgumentException as InvArgExcp;
+use RuntimeException;
+use LogicException;
+
+/**
+ * Field class.
+ *
+ * @package Freyja\Database\Schema
+ * @author Gianluca Merlo <gianluca@squeezyweb.com>
+ * @since 0.1.0
+ * @version 1.0.0
+ */
+class Field {
+  /**
+   * Field name.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var string
+   */
+  private $name;
+
+  /**
+   * Data type.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var string
+   */
+  private $type;
+
+  /**
+   * Data length.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var int
+   */
+  private $length;
+
+  /**
+   * Number of decimals.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var int
+   */
+  private $decimals;
+
+  /**
+   * Default value.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var mixed
+   */
+  private $default;
+
+  /**
+   * Nullable field.
+   *
+   * Whether the field is nullable or not. Default true;
+   *
+   * @since 1.0.0
+   * @access private
+   * @var boolean
+   */
+  private $nullable = true;
+
+  /**
+   * Unsigned data.
+   *
+   * Whether the field is UNSIGNED or not. Only for numeric data types.
+   * Default false.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var boolean
+   */
+  private $unsigned = false;
+
+  /**
+   * Auto increment.
+   *
+   * Whether the field is AUTO_INCREMENT or not. Default false.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var boolean
+   */
+  private $auto_increment = false;
+
+  /**
+   * Types that can be UNSIGNED and AUTO_INCREMENT.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var array
+   */
+  private $allowed = array(
```

Comment:
The name of this property is ambiguous. It would be better to call it something like `$numeric_types`